java/vector/src/main/codegen/templates/UnionListWriter.java

```diff
@@ -178,8 +178,8 @@ public void write(DecimalHolder holder) {
     writer.setPosition(writer.idx()+1);
   }
 
-  public void writeDecimal(int start, ArrowBuf buffer) {
-    writer.writeDecimal(start, buffer);
+  public void writeDecimal(int start, ArrowBuf buffer, ArrowType arrowType) {
```

Comment:
for the changes in public APIs, can you add a deprecated api for the old behavior ?